bundles/org.openhab.binding.goecharger/src/main/java/org/openhab/binding/goecharger/internal/handler/GoEChargerHandler.java

```diff
@@ -0,0 +1,333 @@
+                case PHASES:
+                    int count = 0;
+                    if (goeResponse.energy[4] > 0) { // amps P1
+                        count++;
+                    }
+                    if (goeResponse.energy[5] > 0) { // amps P2
+                        count++;
+                    }
+                    if (goeResponse.energy[6] > 0) { // amps P3
+                        count++;
+                    }
+                    return new DecimalType(count);
+                case TEMPERATURE:
+                    return new QuantityType<>(goeResponse.temperature, SIUnits.CELSIUS);
+                case SESSION_CHARGE_CONSUMPTION:
+                    return new QuantityType<>((Double) (goeResponse.sessionChargeConsumption / 360000d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case SESSION_CHARGE_CONSUMPTION_LIMIT:
+                    return new QuantityType<>((Double) (goeResponse.sessionChargeConsumptionLimit / 10d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case TOTAL_CONSUMPTION:
+                    return new QuantityType<>((Double) (goeResponse.totalChargeConsumption / 10d),
+                            SmartHomeUnits.KILOWATT_HOUR);
+                case FIRMWARE:
+                    return new StringType(goeResponse.firmware);
+                case VOLTAGE_L1:
+                    return new QuantityType<>(goeResponse.energy[0], SmartHomeUnits.VOLT);
+                case VOLTAGE_L2:
+                    return new QuantityType<>(goeResponse.energy[1], SmartHomeUnits.VOLT);
+                case VOLTAGE_L3:
+                    return new QuantityType<>(goeResponse.energy[2], SmartHomeUnits.VOLT);
+                case CURRENT_L1:
+                    return new QuantityType<>((Double) (goeResponse.energy[4] / 10d), SmartHomeUnits.AMPERE);
+                case CURRENT_L2:
+                    return new QuantityType<>((Double) (goeResponse.energy[5] / 10d), SmartHomeUnits.AMPERE);
+                case CURRENT_L3:
+                    return new QuantityType<>((Double) (goeResponse.energy[6] / 10d), SmartHomeUnits.AMPERE);
+                case POWER_L1:
+                    return new QuantityType<>((Double) (goeResponse.energy[7] / 10d),
+                            SmartHomeUnits.WATT);
+                case POWER_L2:
+                    return new QuantityType<>((Double) (goeResponse.energy[8] / 10d),
+                            SmartHomeUnits.WATT);
+                case POWER_L3:
+                    return new QuantityType<>((Double) (goeResponse.energy[9] / 10d),
+                            SmartHomeUnits.WATT);
+            }
+        }
+        return UnDefType.UNDEF;
+    }
+
+    @Override
+    @SuppressWarnings("unchecked")
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        if (command instanceof RefreshType) {
```

Comment:
I would remove that `if` and check the type for each channel. See my comment below.